### ISSUE TITLE
tests != docs.

### ISSUE THREAD

**Issue by Caes95** [hooklift/node-libvirt#73]
It becomes very complicated start to use and contribute to this project, because it lacks documentation, apparently referring to unit testing, as if it were an appropriate way to start, but they are not.

Maybe for beginners like me it is complicated, but the idea is that everyone can contribute, or at least try.

**Comment by mbroadst** [comment]
@Caes95 there is an API page linked to on the front page where others before you have contributed time to start putting documentation together. This is open source, and therefore a best-effort situation and I very much encourage you to help us with this if you continue to learn how to use this module.

The idea behind the tests being helpful is that they are indeed examples of how the module would be used in the real world, and in fact do document the system. I think a really great start would be for you to better describe what kind of documentation you were looking for instead.

Thanks!

**Comment by ghost** [comment]
Docs lack examples. A great start would be basic app setup both local and remote.

**Comment by c4milo** [comment]
@dff3bc, yes, we should have some examples in the documentation, any contributions are very welcome. In the mean time, the best we have to offer are our tests which exercise the API to a reasonable degree.

**Comment by ghost** [comment]
I wouldn't mind documenting this but working off of your test cases is cumbersome at best. If there were just an examples document with basic use of each function then documentation wouldn't be a problem and your project would be a lot more popular. Maybe I'm a noob but trying to use this without the docs is painful, especially since I have a strong desire to use this lib in my project.

**Comment by dustinaevans** [comment]
The last comment was my other profile. Disregard. I just figured it out. I will start putting docs together.

**Comment by c4milo** [comment]
Awesome, thanks @dustinaevans!

**Comment by musharaf780** [comment]
Anyone, please help me o solve this.

Discuss the significance of Agility and Multitenancy in cloud computing.
You are a system administrator of a company whose current IT setup comprises of the following
on-premises services
1. Company’s website
2. CRM Application
3. Microsoft exchange
4. Sharepoint
5. Active Directory Domain Services
6. Storage
Company wants to move all their services on to the cloud except storage and domain services.
Company wants all their data to be stored locally. Propose a best solution for the company that
meets their requirement. The solution would include deployment models (Public, Private and
Hybrid) and all service models (SaaS, PaaS and IaaS). Name which service provider (AWS,
Azure etc.) you will introduce. You have to decide what combination would work best for the
company and justify your solution with proper reasoning
Marks will be graded based on the quality of a solution. Best (economical and flexible) solution
will get the highest marks.
